src/gretel_trainer/models.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from typing import Union
+
+from gretel_client.projects.models import read_model_config
+
+
+class _BaseConfig:
+    """This class should not be used directly, models should be
+    derived from this class
+    """
+
+    _max_header_clusters: int
+    """This should be overridden on concrete classes"""
+
+    # Default values for all configs, may be overridden if needed
+    _max_rows: int = 1000000
```

Comment:
Nit: I think naming these fields something like `_max_header_clusters_limit` and `_max_rows_limit` would make it a bit more readable, for now it's very similar to `max_header_clusters` and it's hard to tell why one has `_` and the other doesn't.